src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,165 @@
+        anotherValidLogin = "AcceptableLogin";
+        thirdValidLogin = "MyName";
+        invalidLogin = "Name";
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void registerDifferentGrownUps_Ok() {
+        john.setAge(ADULT_AGE);
+        john.setLogin(validLogin);
+        john.setPassword(validPassword);
+        maria.setAge(ANOTHER_ADULT_AGE);
+        maria.setLogin(anotherValidLogin);
+        maria.setPassword(anotherValidPassword);
+        User actual = registrationService.register(john);
+        User nextUserToAdd = registrationService.register(maria);
+        assertEquals(actual, john);
+        assertEquals(nextUserToAdd, maria);
+    }
+
+    @Test
+    void registerNotAdult_NotOk() {
+        john.setAge(TEEN_AGE);
+        john.setLogin(validLogin);
+        john.setPassword(validPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+    }
+
+    @Test
+    void inadmissibleAge_NotOK() {
+        john.setAge(null);
+        john.setLogin(validLogin);
+        john.setPassword(validPassword);
+        george.setAge(ZERO_AGE);
+        george.setLogin(anotherValidLogin);
+        george.setPassword(anotherValidPassword);
+        jane.setAge(NEGATIVE_AGE);
+        jane.setLogin(thirdValidLogin);
+        jane.setPassword(validPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+        assertThrows(InvalidDataException.class, () -> registrationService.register(george));
+        assertThrows(InvalidDataException.class, () -> registrationService.register(jane));
+    }
+
+    @Test
+    void shortLogin_NotOk() {
+        john.setAge(ADULT_AGE);
+        john.setLogin(invalidLogin);
+        john.setPassword(validPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+    }
+
+    @Test
+    void shortPassword_NotOk() {
+        john.setAge(ANOTHER_ADULT_AGE);
+        john.setLogin(validLogin);
+        john.setPassword(invalidPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+    }
+
+    @Test
+    void emptyLogin_NotOk() {
+        john.setAge(ADULT_AGE);
+        john.setLogin("");
+        john.setPassword(validPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+    }
+
+    @Test
+    void emptyPassword_NotOK() {
+        john.setAge(ADULT_AGE);
+        john.setLogin(validLogin);
+        john.setPassword("");
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+    }
+
+    @Test
+    void nullLogin_NotOk() {
+        john.setAge(ANOTHER_ADULT_AGE);
+        john.setLogin(null);
+        john.setPassword(validPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+    }
+
+    @Test
+    void nullPassword_NotOk() {
```

Comment:
The name of the test method 'nullPassword_NotOk' does not follow the convention `<methodUnderTest>_<state>_<expectedBehavior>`. Consider renaming it to 'register_nullPassword_NotOk' to reflect the method being tested, the state of the input, and the expected outcome.